From: Jakub Adam <jakub.adam@ktknet.cz>
Date: Sat, 22 Dec 2012 21:46:13 +0100
Subject: fix-ddms-script-paths

---
 ddms/app/etc/ddms |   27 ++-------------------------
 1 file changed, 2 insertions(+), 25 deletions(-)

diff --git a/ddms/app/etc/ddms b/ddms/app/etc/ddms
index d698ec3..8390aff 100755
--- a/ddms/app/etc/ddms
+++ b/ddms/app/etc/ddms
@@ -34,7 +34,7 @@ prog="${progdir}"/`basename "${prog}"`
 cd "${oldwd}"
 
 jarfile=ddms.jar
-frameworkdir="$progdir"
+frameworkdir="/usr/share/java"
 libdir="$progdir"
 if [ ! -r "$frameworkdir/$jarfile" ]
 then
@@ -77,33 +77,10 @@ fi
 
 jarpath="$frameworkdir/$jarfile:$frameworkdir/swtmenubar.jar"
 
-# Figure out the path to the swt.jar for the current architecture.
-# if ANDROID_SWT is defined, then just use this.
-# else, if running in the Android source tree, then look for the correct swt folder in prebuilt
-# else, look for the correct swt folder in the SDK under tools/lib/
-swtpath=""
-if [ -n "$ANDROID_SWT" ]; then
-    swtpath="$ANDROID_SWT"
-else
-    vmarch=`${javaCmd} -jar "${frameworkdir}"/archquery.jar`
-    if [ -n "$ANDROID_BUILD_TOP" ]; then
-        osname=`uname -s | tr A-Z a-z`
-        swtpath="${ANDROID_BUILD_TOP}/prebuilts/tools/${osname}-${vmarch}/swt"
-    else
-        swtpath="${frameworkdir}/${vmarch}"
-    fi
-fi
-
-if [ ! -d "$swtpath" ]; then
-    echo "SWT folder '${swtpath}' does not exist."
-    echo "Please export ANDROID_SWT to point to the folder containing swt.jar for your platform."
-    exit 1
-fi
-
 echo "The standalone version of DDMS is deprecated."
 echo "Please use Android Device Monitor (tools/monitor) instead."
 exec "$javaCmd" \
     -Xmx256M $os_opts $java_debug \
     -Dcom.android.ddms.bindir="$progdir" \
-    -classpath "$jarpath:$swtpath/swt.jar" \
+    -classpath "$jarpath" \
     com.android.ddms.Main "$@"
