From 58bd8bf92acdb6538e5565cfdb14d9a3bf5d6ae4 Mon Sep 17 00:00:00 2001
From: Jakub Adam <jakub.adam@tieto.com>
Date: Fri, 15 Mar 2013 12:11:29 +0100
Subject: [PATCH] fix-ddms-script-paths

---
 ddms/app/etc/ddms |   27 ++-------------------------
 1 file changed, 2 insertions(+), 25 deletions(-)

diff --git a/ddms/app/etc/ddms b/ddms/app/etc/ddms
index 79b93f9..b27b8d9 100755
--- a/ddms/app/etc/ddms
+++ b/ddms/app/etc/ddms
@@ -34,7 +34,7 @@ prog="${progdir}"/`basename "${prog}"`
 cd "${oldwd}"
 
 jarfile=ddms.jar
-frameworkdir="$progdir"
+frameworkdir="/usr/share/java"
 libdir="$progdir"
 if [ ! -r "$frameworkdir/$jarfile" ]
 then
@@ -77,29 +77,6 @@ fi
 
 jarpath="$frameworkdir/$jarfile:$frameworkdir/swtmenubar.jar"
 
-# Figure out the path to the swt.jar for the current architecture.
-# if ANDROID_SWT is defined, then just use this.
-# else, if running in the Android source tree, then look for the correct swt folder in prebuilt
-# else, look for the correct swt folder in the SDK under tools/lib/
-swtpath=""
-if [ -n "$ANDROID_SWT" ]; then
-    swtpath="$ANDROID_SWT"
-else
-    vmarch=`${javaCmd} -jar "${frameworkdir}"/archquery.jar`
-    if [ -n "$ANDROID_BUILD_TOP" ]; then
-        osname=`uname -s | tr A-Z a-z`
-        swtpath="${ANDROID_BUILD_TOP}/prebuilts/tools/${osname}-${vmarch}/swt"
-    else
-        swtpath="${frameworkdir}/${vmarch}"
-    fi
-fi
-
-if [ ! -d "$swtpath" ]; then
-    echo "SWT folder '${swtpath}' does not exist."
-    echo "Please export ANDROID_SWT to point to the folder containing swt.jar for your platform."
-    exit 1
-fi
-
 if [ -x $progdir/monitor ]; then
     echo "The standalone version of DDMS is deprecated."
     echo "Please use Android Device Monitor (tools/monitor) instead."
@@ -107,5 +84,5 @@ fi
 exec "$javaCmd" \
     -Xmx256M $os_opts $java_debug \
     -Dcom.android.ddms.bindir="$progdir" \
-    -classpath "$jarpath:$swtpath/swt.jar" \
+    -classpath "$jarpath" \
     com.android.ddms.Main "$@"
-- 
1.7.10.4

